### PR TITLE
Fix log message for auth token creation

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Fixes the condition to print success/error message for creating a new auth token. This was introduced in https://github.com/DataDog/datadog-agent/pull/10079

### Describe how to test/QA your changes

* Ensure `/etc/datadog-agent/auth_token` doesn't exist.
* Start the agent
* Ensure logs contain `Wrote auth token acl` instead of `Failed to write auth token acl %!s(<nil>)`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/api/security/security.go [hunk 1]
```diff
@@ -244,7 +244,7 @@ func saveAuthToken(token, tokenPath string) error {
 		return err
 	}
 
-	if err := perms.RestrictAccessToUser(tokenPath); err != nil {
+	if err := perms.RestrictAccessToUser(tokenPath); err == nil {
```
Comment (paulcacheux) [inline pkg/api/security/security.go:247]: This change is good, but wouldn't be more idiomatic to actually swap the bodies of the if and else branches ? I think we rarely see `err == nil` in any code.

Comment (paulcacheux) [reply pkg/api/security/security.go:247]: Otherwise thank you for the investigation and for this PR

Comment (bkabrda) [reply pkg/api/security/security.go:247]: Good point, let me do that :)

**Review by remeh — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

File: pkg/api/security/security.go (hunk 1 above)
Comment (bkabrda) [reply pkg/api/security/security.go]: @paulcacheux fixed

**Review by lebauce — APPROVED** [review]

**Pull request merged**